### PR TITLE
Don't modify user-schemas in Material Object Renderer

### LINKED ISSUE #1712: Don't modify user-schemas in Material Object Renderer
**Description**
A detail ui schema can be set for objects. The Material Object renderer should not modify user-given ui schemas.

**To Reproduce**
1. Specify a detail ui schema with a label for an object control
2. The Material Object renderer will modify the label when the object control is not on the root path

**Expected behavior**
The Material Object renderer should not modify user-given ui schemas.

**Used Setup:**
 - Framework: [react]
 - RendererSet: [material]

**Additional context**
Follow up of #1701

### PR DESCRIPTION
**Pull request by LukasBoll**
Don't modify user-schemas in Material Object Renderer

closes #1712

Signed-off-by: Lukas Boll lukas-bool@web.de

### CODE REVIEW

File: packages/material/src/complex/MaterialObjectRenderer.tsx [hunk 1]
```diff
@@ -47,23 +47,22 @@ export const MaterialObjectRenderer = ({
...
+
   const detailUiSchema = useMemo(
     () =>
       findUISchema(
         uischemas,
         schema,
         uischema.scope,
         path,
-        'Group',
+        ()=>Generate.uiSchema(schema, 'Group',undefined,undefined,label),
```
Comment (max-elia) [inline packages/material/src/complex/MaterialObjectRenderer.tsx:58]: I suggest a consistent use of spaces.
```suggestion
        () => Generate.uiSchema(schema, 'Group', undefined, undefined, label),
```

File: packages/core/src/generators/uischema.ts [hunk 2]
```diff
@@ -211,9 +216,10 @@ export const generateDefaultUISchema = (
   jsonSchema: JsonSchema,
   layoutType = 'VerticalLayout',
   prefix = '#',
-  rootSchema = jsonSchema
+  rootSchema = jsonSchema,
+  label?: string
 ): UISchemaElement =>
   wrapInLayoutIfNecessary(
-    generateUISchema(jsonSchema, [], prefix, '', layoutType, rootSchema),
+    generateUISchema(jsonSchema, [], prefix, '', layoutType, rootSchema,label),
```
Comment (max-elia) [inline packages/core/src/generators/uischema.ts:223]:
```suggestion
    generateUISchema(jsonSchema, [], prefix, '', layoutType, rootSchema, label),
```
Consistent spacing

**Review by max-elia — APPROVED** [review]
The changes LGTM.
Tested it and it works as expected.

**Comment by LukasBoll** [comment]
Thanks! I updated the code accordingly.

File: packages/core/src/generators/Generate.ts [hunk 3]
```diff
@@ -33,7 +33,8 @@ export const Generate: {
     jsonSchema: JsonSchema,
     layoutType?: string,
     prefix?: string,
-    rootSchema?: JsonSchema
+    rootSchema?: JsonSchema,
+    label?: string
```
Comment (sdirix) [inline packages/core/src/generators/Generate.ts:37]: Can we remove this again?

File: packages/material/src/complex/MaterialObjectRenderer.tsx [hunk 4]
```diff
@@ -47,23 +47,22 @@ export const MaterialObjectRenderer = ({
...
+
   const detailUiSchema = useMemo(
     () =>
       findUISchema(
         uischemas,
         schema,
         uischema.scope,
         path,
-        'Group',
+        () => Generate.uiSchema(schema, 'Group', undefined, undefined, label),
```
Comment (sdirix) [inline packages/material/src/complex/MaterialObjectRenderer.tsx:58]: I would prefer assigning the label as part of the generator

**Review by sdirix — CHANGES_REQUESTED** [review]

File: packages/material/src/complex/MaterialObjectRenderer.tsx [hunk 5]
```diff
@@ -47,23 +48,22 @@ export const MaterialObjectRenderer = ({
...
         path,
-        'Group',
+        () =>  {return ({...Generate.uiSchema(schema, 'Group'), label: label} as GroupLayout)},
         uischema,
         rootSchema
       ),
     [uischemas, schema, uischema.scope, path, uischema, rootSchema]
   );
   if (isEmpty(path)) {
     detailUiSchema.type = 'VerticalLayout';
```
Comment (sdirix) [inline packages/material/src/complex/MaterialObjectRenderer.tsx:66]: We should also remove this modification ;)

File: packages/core/src/reducers/reducers.ts [hunk 6]
```diff
@@ -67,24 +67,28 @@ export const jsonFormsReducerConfig = {
  * @param schema the JSON schema describing the data to be rendered
  * @param schemaPath the according schema path
  * @param path the instance path
- * @param fallbackLayoutType the type of the layout to use
+ * @param fallback the type of the layout to use or a schema-generator function
```
Comment (sdirix) [inline packages/core/src/reducers/reducers.ts:70]:
```suggestion
 * @param fallback the type of the layout to use or a UI-schema-generator function
```

File: packages/material/src/complex/MaterialObjectRenderer.tsx [hunk 7]
```diff
@@ -54,17 +54,12 @@ export const MaterialObjectRenderer = ({
         schema,
         uischema.scope,
         path,
-        'Group',
+        () =>  {return isEmpty(path)?Generate.uiSchema(schema, 'VerticalLayout'):{...Generate.uiSchema(schema, 'Group'), label: label}},
```
Comment (sdirix) [inline packages/material/src/complex/MaterialObjectRenderer.tsx:57]:
```suggestion
        () => isEmpty(path) ? Generate.uiSchema(schema, 'VerticalLayout') : {...Generate.uiSchema(schema, 'Group'), label},
```

File: packages/material/src/complex/MaterialObjectRenderer.tsx [hunk 8]
```diff
@@ -54,17 +54,12 @@ export const MaterialObjectRenderer = ({
...
         uischema,
         rootSchema
       ),
     [uischemas, schema, uischema.scope, path, uischema, rootSchema]
```
Comment (sdirix) [inline packages/material/src/complex/MaterialObjectRenderer.tsx:61]: When the label changes we might need to generate a new UI Schema, therefore the label needs to be added to the dependency array
```suggestion
    [uischemas, schema, uischema.scope, path, label, uischema, rootSchema]
```

**Comment by LukasBoll** [comment]
Thank you, I added your suggestion!

**Review by sdirix — APPROVED** [review]

**Pull request merged**